src/base/OpenMCCellAverageProblem.C

```diff
@@ -442,6 +448,32 @@ OpenMCCellAverageProblem::OpenMCCellAverageProblem(const InputParameters & param
   }
 }
 
+const MooseMesh &
+OpenMCCellAverageProblem::getMooseMesh() const
+{
+  if (_use_displaced && !_displaced_problem && !_first_transfer)
+    mooseWarning("Displaced mesh was requested but the displaced problem does not exist. "
```

Comment:
Please add a test that triggers this warning, so we're sure it does get displayed to users as expected. Your test specs will look like following. The `cli_args` allows you to write which command-line parameters you want to run with your test. `--error` will cause all warnings to become errors so that you can catch them.

```
[test]
  type = RunException
  input = my_input.i
  cli_args = '--error'
  expect_err = "Displaced mesh was requested but the displaced problem does not exist."
  required_objects = 'OpenMCCellAverageProblem'
```